### PR TITLE
Day/02/phase1

### PR DESCRIPTION
**Pull request by bcarlier75**
Data for ex05 can be found here:

https://drive.google.com/open?id=1v50zUV6S7VUyZLR_qszDA2c63pHp5F6J

Ask here or slack me @bcarlier if you have any question.

### CODE REVIEW

File: day02/ex10/ex10.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+**Objectives:**
+
+The goal of this exercise is to recreate the function confusion_matrix of sklearn.metrics and to learn what represents
+the confusion matrix.
+
+**Instructions:**
+
+For the sake of simplicity, we will only ask you to have three parameters.
+Be careful to respect the order, true labels are rows and predicted labels are columns !
```
Comment (madvid) [inline day02/ex10/ex10.md]: What do you mean more precisely ? Are you saying that Y_pred is a vector (Nx1) and  Y_true a  matrix of 1xN ?

File: day02/ex01/ex01.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+$$
+J( \theta) = -\frac{1} {m} * \lbrack \sum_{i = 1}^{m} y^{(i)}log(h_{ \theta }(x^{(i)})) + (1 - y^{(i)})log(1 - h_{ \theta }(x^{(i)}))\rbrack
+$$
+
+where:  
+* $J( \theta)$ is the cost function with theta being the weights.
+* $m$ is the length of $y$, i.e. the number of observations in our sample.
+* $h_{\theta}(x)$ also called y_pred, is the calculated hypothesis and it represents the predicted output (formula below).
+* $y$, also called y_true, represents the desired output, either 1 or 0.
```
[the remark this answers is not in the document]
Comment (bcarlier75) [reply day02/ex01/ex01.md]: Ou ca deux fois "also called y_pred" ?
sur ligne 25 : "also called y_pred"
sur ligne 26: "also called y_true"

File: day02/ex04/ex04.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        y_pred: a scalar or a numpy ndarray for the predicted labels
+    Returns: 
+        The gradient as a scalar or a numpy ndarray of the width of x.
+        None on any error.
+    Raises:
+        This function shouldn't raise any Exception.
+    """
+```
+  
+* x length (x.shape[0] if it's a 2d ndarray) should match y_true and y_pred length, i.e. we should have the same
```
[the remark this answers is not in the document]
Comment (bcarlier75) [reply day02/ex04/ex04.md]: pour le "it's" ?

File: day02/ex10/ex10.md (hunk 1 above)
Comment (bcarlier75) [reply day02/ex10/ex10.md]: What I mean is this, I will add this code sniplet :
```
#                      [predicted labels]    
#                       label_1  label_2
#   [ true ]  label_1         .        .
#   [labels]  label_2         .        .
```
Check [this link.](https://miro.medium.com/max/1194/0*wKaznIJzZF54b87B.jpg)

File: day02/ex01/ex01.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+This function is called the Cross-Entropy loss or logistic loss.
+We encourage you to get a look at
+[this section](https://en.wikipedia.org/wiki/Cross_entropy#Cross-entropy_error_function_and_logistic_regression)
+of the Cross entropy Wikipedia.
+
+Formula for hypothesis:
+
+$$
+\hat{y} = h_{ \theta }(x) = \frac{1} {1 + e^{-\theta \cdot x}}
```
[the remark this answers is not in the document]
Comment (bcarlier75) [reply day02/ex01/ex01.md]: Je suis parti sur un vecteur ligne

File: day02/ex01/ex01.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+$$
+
+As you may have noticed, this is our sigmoid function with $\theta \cdot x$ as argument.
```
[the remark this answers is not in the document]
Comment (bcarlier75) [reply day02/ex01/ex01.md]: En effet, pour un vecteur ligne ca marche eheh 👍

File: day02/ex02/ex02.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+You must implement the following formula as a function:  
+ 
+ $$
+{\displaystyle {\overrightarrow {\nabla }}} = \sum_{i = 1}^{m} (h_\theta (x^{(i)}) - y^{(i)}) \cdot x^{(i)} 
+ $$
+ 
+where:
+* ${\displaystyle {\overrightarrow {\nabla }}}$ is the gradient
+* $x$ are the input coefficients
```
[the remark this answers is not in the document]
Comment (bcarlier75) [reply day02/ex02/ex02.md]: Je dirais 'are the variables of your models'

File: day02/ex05/ex05.md [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+learning. I would strongly advise you to put it on your top list of sites you have to visit everyday if you wish to work
+on a data related position in the future. PS: private navigation in your browser is the way to go for unlimited reading)
+
+**Regarding both datasets:**
+
+* The 1st column correspond to the 'income' variable : 0 if '<=50K' and 1 if '>50k'. 
+* The 2nd to the 6th columns correspond to the standardized numerical variables : 'age', 'education-num', 
+'capital-gain', 'capital-loss' and 'hours-per-week'.
+* The 7th to the 81th columns correspond to dummy variables for the following categorical variables : 'workClass', 
+'marital-status', 'occupation', 'relationship', 'race', 'sex' and 'native-country'.
```
[the remark this answers is not in the document]
Comment (bcarlier75) [reply day02/ex05/ex05.md]: That's an other way to do this, we should discuss about this indeed. For now it should be ok to leave it as is it. We would need to have it harmonized before PRing into master though.

**Comment by Talasta** [comment]
Merci de merge dans dev et de delete la branche.

**Comment by ppeigne** [comment]
Done 🔥
